recipes/halide/build.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# On OSX, we need to ensure we're using conda's gcc/g++
+if [[ `uname` == Darwin ]]; then
+     export CC=gcc
+     export LD=gcc
+     export CXX=g++
```

Comment:
I think that's right. Will look at the thrift recipe PR. 

For the time being, it looks like Halide requires a newer version of LLVM (3.5): Can't find llvm or version of llvm too old (we need 3.5 or greater):